core/domain/activity_jobs_one_off.py

```diff
@@ -471,3 +478,129 @@ def map(snapshot_model):
     def reduce(key, values):
         """Implements the reduce function for this job."""
         yield (key, len(values))
+
+
+class ValidateSnapshotMetadataModelsJob(jobs.BaseMapReduceOneOffJobManager):
+    """Job that validates whether each SnapshotMetadata model has a
+    corresponding CommitLog model pair and the corresponding parent model.
+    """
+
+    SnapshotMetadataModels = [
+        collection_models.CollectionSnapshotMetadataModel,
+        collection_models.CollectionRightsSnapshotMetadataModel,
+        exp_models.ExplorationSnapshotMetadataModel,
+        exp_models.ExplorationRightsSnapshotMetadataModel,
+        question_models.QuestionSnapshotMetadataModel,
+        skill_models.SkillSnapshotMetadataModel,
+        story_models.StorySnapshotMetadataModel,
+        topic_models.TopicSnapshotMetadataModel,
+        subtopic_models.SubtopicPageSnapshotMetadataModel,
+        topic_models.TopicRightsSnapshotMetadataModel
+    ]
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        """Return a list of SnapshotMetadata models that is associated
+        with a CommitLogEntry model.
+        """
+        return ValidateSnapshotMetadataModelsJob.SnapshotMetadataModels
+
+    @staticmethod
+    def map(snapshot_model):
+        """Implements the map function for this job."""
+        class_name = snapshot_model.__class__.__name__
+        missing_commit_log_msg = 'MISSING COMMIT LOGS'
+        # This list consists of the rights snapshot metadata models for which
+        # the commit log model is not created when the commit cmd is create
```

Comment:
Put quotes around "create" and "delete".

Reply:
Done.